pkg/api/controllers/usercontrollers/usercontroller.go

```diff
@@ -67,12 +69,19 @@ type userControllersController struct {
 	clustered     bool
 	manager       *clustermanager.Manager
 	clusterLister v3.ClusterLister
+	clusters      v3.ClusterInterface
 	ctx           context.Context
 	peers         tpeermanager.Peers
 	start         time.Time
 }
 
 func (u *userControllersController) sync(key string, cluster *v3.Cluster) error {
+	if cluster != nil && cluster.ObjectMeta.DeletionTimestamp != nil {
```

Comment:
nit: `cluster.DeletionTimestamp` should be enough to access.

Reply:
Good call, updated.